### ISSUE TITLE
v0.54 500 error from sql query in custom template; code worked in v0.53; found a workaround

### ISSUE THREAD

**Issue by jrdmb** [simonw/datasette#1209]
v0.54 500 error in sql query template; code worked in v0.53; found a workaround

**schema:**   
CREATE TABLE "talks" ("talk" TEXT,"series" INTEGER, "talkdate" TEXT)  
CREATE TABLE "series" ("id" INTEGER PRIMARY KEY, "series" TEXT, talks_list TEXT default '', website TEXT default '');

**Live example of correctly rendered template in v.053:** https://cosmotalks-cy6xkkbezq-uw.a.run.app/cosmotalks/talks/1

**Description of problem:** I needed 'sql select code' in a custom row-mydatabase-mytable.html template to lookup the series name for a foreign key integer value in the talks table.  So `metadata.json` specifies the `datasette-template-sql` plugin.

The code below worked perfectly in v0.53 (just the relevant sql statement part is shown):

```
{# custom addition #}  
{% for row in display_rows %}  
    ...  
    {% set sname = sql("select series from series where id = ?", [row.series]) %}  
    <strong>Series name: {{ sname[0].series }}  
    ...  
{% endfor %}  
{# End of custom addition #}  
```

**In v0.54, that code resulted in a 500 error with a 'no such table series' message.**  A second query in that template also did not work but the above is fully illustrative of the problem.

All templates were up-to-date along with datasette v0.54.

After fiddling around with trying different things, what worked was the syntax from [Querying a different database from the datasette-template-sql github repo](https://github.com/simonw/datasette-template-sql#querying-a-different-database) to add the database name to the sql statement:

`{% set sname = sql("select series from series where id = ?", [row.series], database="cosmotalks") %}`

Though this was found to work, it should not be necessary to add `database="cosmotalks"` since we're not querying a different database, just a table within the same database.

**Comment by simonw** [comment]
Good catch on the workaround here. The root problem is that `datasette-template-sql` looks for the first available databsae if you don't provide it with a `database=` argument, and in Datasette 0.54 the first available database changed to being the new `_internal` database.

Is this a bug? I think it is - because the documented behaviour on https://docs.datasette.io/en/stable/internals.html#get-database-name is this:

> `name` - string, optional
>
> The name to be used for this database - this will be used in the URL path, e.g. `/dbname`. If not specified Datasette will pick one based on the filename or memory name.

Since the new behaviour differs from what was in the documentation I'm going to treat this as a bug and fix it.

**Comment by simonw** [comment]
Oh weird, the code is supposed to be doing that already: https://github.com/simonw/datasette/blob/382e9ecd1d429102417b17a1bd75f066cb904e24/datasette/app.py#L367-L371

Here's the associated test: https://github.com/simonw/datasette/blob/dde3c500c73ace33529672f7d862b76753d309cc/tests/test_internals_datasette.py#L21-L24

**Comment by simonw** [comment]
I'm pretty sure this is a bug in `datasette-template-sql` itself - moving it there.
